sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -244,7 +244,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   def json(path: String): DataFrame = {
     // This method ensures that calls that explicit need single argument works, see SPARK-16009
-    json(Seq(path): _*)
+    option("path", path).json(Seq.empty: _*)
```

Comment:
@gatorsmile setting `path` or `paths` are both fine for json, parquet, and other file formats. See https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L312

You don't need to change this file.